test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -87,32 +89,49 @@ def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
         local_weight_narrowed = local_linear.weight.narrow(
             sharded_dim, start_pos, chunk_size
         )
+        local_bias_grad = local_linear.bias.grad
+        sharded_bias_grad = sharded_linear.bias.grad
+        # For col-wise sharding, the gradient of bias is also sharded.
+        if sharded_dim == 0:
+            dist.all_reduce(local_bias_grad)
+            local_bias_grad = local_bias_grad.narrow(
+                sharded_dim, start_pos, chunk_size
+            )
+            sharded_bias_grad = sharded_bias_grad.narrow(
+                sharded_dim, start_pos, chunk_size
+            )
```

Comment:
This would be unexpected from the users point of view. The bias is not sharded, yet the grad of the bias received by the user is sharded.

I think we should narrow and apply the bias using an autograd function. The forward part of that autograd function would be exactly what we do in this PR where we narrow the bias and add it. In the backward pass we would gather the grad of the biases using this logic above. That way the user always gets a non-sharded grad for the bias.

Reply:
sure, sounds good!